doc/rest-api.md

```diff
@@ -1818,3 +1818,308 @@ Input (none at present):
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+ Input:
+
+    {
+        "config": {
+            "size": "15032385536",
+            "source": "pool1",
+            "volume.block.filesystem": "xfs",
+            "volume.block.mount_options": "discard",
+            "volume.lvm.thinpool_name": "LXDPool",
+            "volume.size": "10737418240"
+        }
+    }
+
+### PATCH
+ * Description: update the storage pool configuration
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {
+            "volume.block.filesystem": "xfs",
+        }
+    }
+
+### DELETE
+ * Description: delete a storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input (none at present):
+
+    {
+    }
+
+## /1.0/storage-pools/<name>/volumes
+### GET
+
+ * Description: list of storage volumes
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of storage volumes that currently exist on a given storage pool
+
+    [
+        {
+            "type": "container",
+            "used_by": [],
+            "name": "alp1/snap1",
+            "config": {}
+        },
+        {
+            "type": "container",
+            "used_by": [],
+            "name": "alp1/snap2",
+            "config": {}
+        },
+    ]
+
+## /1.0/storage-pools/<pool>/volumes
+### GET
+ * Description: list all storage volumes on a storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+    {
+        "type": "sync",
+        "status": "Success",
+        "status_code": 200,
+        "error_code": 0,
+        "error": "",
+        "metadata": [
+            {
+                "type": "container",
+                "used_by": [],
+                "name": "alp1",
+                "config": {
+                "size": "0"
+                }
+            },
+            {
+                "type": "container",
+                "used_by": [],
+                "name": "alp1/snap0",
+                "config": {
+                    "size": "0"
+                }
+            },
+            {
+                "type": "image",
+                "used_by": [],
+                "name": "ade3a9bcd7ba27456673611304238c424ced1772f69d7c6b031356831d94e8ee",
+                "config": {
+                    "size": "0"
+                }
+            },
+            {
+                "type": "custom",
+                "used_by": [],
+                "name": "bla",
+                "config": {
+                    "size": "0"
+                }
+            }
+        ]
+    }
```

Comment:
done
Currently, storage volumes are always reported with urls relative to the storage pool:

```
        "/1.0/storage-pools/pool1/volumes/containers/alp1",
        "/1.0/storage-pools/pool1/volumes/containers/alp1/snap0",
        "/1.0/storage-pools/pool1/volumes/custom/bla"
```

Let me know if you'd prefer a different format.